tests/test-softhsm.sh

```diff
@@ -0,0 +1,171 @@
+    echo ${PINVALUE} > ${PINFILE}
+
+    # init
+    softhsm2-util --init-token --label $OBJNAME --free --pin $PINVALUE --so-pin 12345678
+
+    # generate RSA key pair
+    KEYID='0001'
+    URIKEYID="%00%01"
+    pkcs11-tool --keypairgen --key-type="RSA:2048" --login --pin=$PINVALUE --module="$P11LIB" --label="RSA" --id="$KEYID"
+
+    BASEURIWITHPIN="pkcs11:id=${URIKEYID};pin-value=${PINVALUE}"
+    BASEURI="pkcs11:id=${URIKEYID}"
+    PUBURI="pkcs11:type=public;id=${URIKEYID}"
+    PRIURI="pkcs11:type=private;id=${URIKEYID}"
+
+    title LINE "RSA PKCS11 URIS"
+    echo "${BASEURIWITHPIN}"
+    echo "${BASEURI}"
+    echo "${PUBURI}"
+    echo "${PRIURI}"
+    echo ""
+
+    # generate ECC key pair
+    KEYID='0002'
+    URIKEYID="%00%02"
+    pkcs11-tool --keypairgen --key-type="EC:secp256r1" --login --pin=$PINVALUE --module="$P11LIB" --label="ECC" --id="$KEYID"
+
+    ECBASEURIWITHPIN="pkcs11:id=${URIKEYID};pin-value=${PINVALUE}"
+    ECBASEURI="pkcs11:id=${URIKEYID}"
+    ECPUBURI="pkcs11:type=public;id=${URIKEYID}"
+    ECPRIURI="pkcs11:type=private;id=${URIKEYID}"
+
+    KEYID='0003'
+    URIKEYID="%00%03"
+    pkcs11-tool --keypairgen --key-type="EC:secp256r1" --login --pin=$PINVALUE --module="$P11LIB" --label="PeerECC" --id="$KEYID"
+
+    ECPEERBASEURIWITHPIN="pkcs11:id=${URIKEYID};pin-value=${PINVALUE}"
+    ECPEERBASEURI="pkcs11:id=${URIKEYID}"
+    ECPEERPUBURI="pkcs11:type=public;id=${URIKEYID}"
+    ECPEERPRIURI="pkcs11:type=private;id=${URIKEYID}"
+
+    title LINE "EC PKCS11 URIS"
+    echo "${ECBASEURIWITHPIN}"
+    echo "${ECBASEURI}"
+    echo "${ECPUBURI}"
+    echo "${ECPRIURI}"
+    echo "${ECPEERBASEURIWITHPIN}"
+    echo "${ECPEERBASEURI}"
+    echo "${ECPEERPUBURI}"
+    echo "${ECPEERPRIURI}"
+    echo ""
+
+    title PARA "Show contents of softhsm token"
+    echo " ----------------------------------------------------------------------------------------------------"
+    pkcs11-tool -O --login --pin=$PINVALUE --module="$P11LIB"
+    echo " ----------------------------------------------------------------------------------------------------"
+
+    title PARA "Start the p11-kit server and check if it works"
+    echo " ----------------------------------------------------------------------------------------------------"
+    # p11-kit complains if there is not runtime directory
+    if [ -z "$XDG_RUNTIME_DIR" ]; then
+        export XDG_RUNTIME_DIR=$PWD
+    fi
+    eval $(p11-kit server --provider "$P11LIB" "pkcs11:")
+    echo " ----------------------------------------------------------------------------------------------------"
+    pkcs11-tool -O --login --pin=$PINVALUE --module="$P11KITCLIENTPATH"
+    echo " ----------------------------------------------------------------------------------------------------"
+
+    export PKCS11_PROVIDER_MODULE=$P11KITCLIENTPATH
+    #export PKCS11_PROVIDER_MODULE=/usr/lib64/pkcs11-spy.so
+    #export PKCS11SPY=$P11KITCLIENTPATH
+    export NOPSS="1"
```

Comment:
I think positively named variables are more understable.

Can you add the following to the common.sh file:
TEST_RSAPSS="1"
TEST_ECC_SHA2="1"
TEST_OAEP_SHA2="1"
TEST_HKDF="1"

then override them in this file by setting them to "0" ?

Reply:
Fine for me. I wanted more to keep this on an exception basis that by default everything works and there would be some exceptions to be skipped, but it turned out the softhsm has problems with more stuff.

Reply:
In the end it seem reasonable to be able to skip any test based on what a token support, I envision adding test for opencryptoki and HW tokens once we have a way to test them, so fine control will be needed anyway I think.